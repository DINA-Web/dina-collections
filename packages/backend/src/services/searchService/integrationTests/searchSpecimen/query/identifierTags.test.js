const {
  describe: apiSampleDescribe,
  hook,
} = require('common/src/testUtilities/envBackendApiSampleData')
const waitForApiRestart = require('../../../../../utilities/test/waitForApiRestart')
const resetElasticSpecimenIndex = require('../../../../../utilities/test/db/resetElasticSpecimenIndex')

const {
  chain,
  createRequestBuilder,
  runTestCases,
  specimenCountEquals,
  tagTypeEquals,
} = require('../../utilities')

apiSampleDescribe(`searchSpecimen - query - tags - identifier`, () => {
  hook(beforeAll, () => {
    return waitForApiRestart().then(() => {
      return resetElasticSpecimenIndex()
    })
  })

  describe(`tag type aggregations`, () => {
    const requestBuilder = createRequestBuilder({
      aggregationFunction: 'aggregateIdentifierTagTypes',
      aggregationType: 'tagTypes',
      filterFunction: 'searchIdentifierTags',
      resource: 'identifierTag',
      tagPath: 'attributes.tags.identifierTags',
    })
    const testCases = [
      {
        aggregate: true,
        compareQueryTypesResult: true,
        expectedCount: 5,
        queryTypes: ['raw', 'dina'],
        testFn: res => {
          const ids = res.data.map(({ id }) => {
            return id
          })

          expect(ids).toEqual([
            'catalog-no',
            'old-skeleton-no',
            'old-skin-no',
            'sva-no',
            'loan-no',
          ])
        },
        title: 'returns all tag types',
      },
    ]
    runTestCases({ requestBuilder, testCases })
  })
  describe(`tag aggregations`, () => {
    const requestBuilder = createRequestBuilder({
      aggregationFunction: 'aggregateIdentifierTagValues',
      aggregationType: 'tagValues',
      filterFunction: 'searchIdentifierTags',
      resource: 'identifierTag',
      tagPath: 'attributes.tags.identifierTags',
    })
    describe(`no filters`, () => {
      const testCases = [
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 35,
          queryTypes: ['raw', 'dina'],
          snapshot: true,
          title:
            'returns all aggregated identifiers when no filters are present',
        },
      ]
      runTestCases({ requestBuilder, testCases })
    })
    describe(`tagType filters`, () => {
      const testCases = [
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 16,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['catalog-no'],
          title: 'returns all aggregated catalog-no identifiers',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 9,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['old-skeleton-no'],
          title: 'returns all aggregated old-skeleton-no identifiers',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 7,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['old-skin-no'],
          title: 'returns all aggregated old-skin-no identifiers',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 2,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['sva-no'],
          title: 'returns all aggregated sva-no identifiers',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 1,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['loan-no'],
          title: 'returns all aggregated loan-no identifiers',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 3,
          queryTypes: ['raw', 'dina'],
          tagTypes: ['loan-no', 'sva-no'],
          title: 'returns all aggregated loan-no and sva-no identifiers',
        },
      ]
      runTestCases({ requestBuilder, testCases })
    })
    describe(`tagValue filters`, () => {
      const testCases = [
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 0,
          queryTypes: ['raw', 'dina'],
          tagValue: '11111',
          title: 'returns 0 matching identifiers for non existing tagValue',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 1,
          queryTypes: ['raw', 'dina'],
          tagValue: '530183',
          title: 'returns 1 matching identifiers for exact matching tagValue',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 1,
          queryTypes: ['raw', 'dina'],
          tagValue: '53018*',
          testFn: chain([
            tagTypeEquals('catalog-no'),
            specimenCountEquals([1]),
          ]),
          title: 'returns 1 matching identifiers for 53018*',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 2,
          queryTypes: ['raw', 'dina'],
          tagValue: '53*',
          testFn: chain([
            tagTypeEquals('catalog-no'),
            specimenCountEquals([1, 1]),
          ]),
          title: 'returns 2 matching identifiers for 53*',
        },
        {
          aggregate: true,
          compareQueryTypesResult: true,
          expectedCount: 9,
          queryTypes: ['raw', 'dina'],
          tagValue: '5*',
          testFn: specimenCountEquals([1, 1, 1, 1, 1, 1, 1, 1, 1]),
          title: 'returns 9 matching identifiers for 5*',
        },
        {
          aggregate: true,
          expectedCount: 25,
          queryTypes: ['raw', 'dina'],
          tagValue: '*5*',
          title: 'returns 25 matching identifiers for *5*',
        },
      ]

      runTestCases({ requestBuilder, testCases })
    })
    describe('tagType and tagValue filters ', () => {
      describe(`tagType: catalog-no`, () => {
        const testCases = [
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 0,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['catalog-no'],
            tagValue: '11111',
            title: 'returns 0 matching identifiers for non existing tagValue',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 7,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['catalog-no'],
            tagValue: '5*',
            testFn: tagTypeEquals('catalog-no'),
            title: 'returns 7 matching identifiers for 5*',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['catalog-no'],
            tagValue: '500001',
            testFn: tagTypeEquals('catalog-no'),
            title: 'returns 1 matching identifiers for 500001',
          },
        ]

        runTestCases({ requestBuilder, testCases })
      })

      describe(`tagType: old-skeleton-no`, () => {
        const testCases = [
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 0,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skeleton-no'],
            tagValue: '11111',
            title: 'returns 0 matching identifiers for non existing tagValue',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skeleton-no'],
            tagValue: '5*',
            testFn: tagTypeEquals('old-skeleton-no'),
            title: 'returns 1 matching identifiers for 1*',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skeleton-no'],
            tagValue: '1,285',
            testFn: tagTypeEquals('old-skeleton-no'),
            title: 'returns 1 matching identifiers for 1,285',
          },
        ]

        runTestCases({ requestBuilder, testCases })
      })

      describe(`tagType: old-skin-no`, () => {
        const testCases = [
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 0,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skin-no'],
            tagValue: '11111',
            title: 'returns 0 matching identifiers for non existing tagValue',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 2,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skin-no'],
            tagValue: '1*',
            testFn: tagTypeEquals('old-skin-no'),
            title: 'returns 2 matching identifiers for 1*',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['old-skin-no'],
            tagValue: '1; 4406; 52',
            testFn: tagTypeEquals('old-skin-no'),
            title: 'returns 1 matching identifiers for 1; 4406; 52',
          },
        ]

        runTestCases({ requestBuilder, testCases })
      })

      describe(`tagType: sva-no`, () => {
        const testCases = [
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 0,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['sva-no'],
            tagValue: '11111',
            title: 'returns 0 matching identifiers for non existing tagValue',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 2,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['sva-no'],
            tagValue: 'v0*',
            testFn: tagTypeEquals('sva-no'),
            title: 'returns 2 matching identifiers for 5*',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['sva-no'],
            tagValue: 'v0253/98',
            testFn: tagTypeEquals('sva-no'),
            title: 'returns 1 matching identifiers for v0253/98',
          },
        ]

        runTestCases({ requestBuilder, testCases })
      })
      describe(`tagType: loan-no`, () => {
        const testCases = [
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 0,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['loan-no'],
            tagValue: '11111',
            title: 'returns 0 matching identifiers for non existing tagValue',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['loan-no'],
            tagValue: '20*',
            testFn: tagTypeEquals('loan-no'),
            title: 'returns 1 matching identifiers for 20*',
          },
          {
            aggregate: true,
            compareQueryTypesResult: true,
            expectedCount: 1,
            queryTypes: ['raw', 'dina'],
            tagTypes: ['loan-no'],
            tagValue: '2012-21',
            testFn: tagTypeEquals('loan-no'),
            title: 'returns 1 matching identifiers for 2012-21',
          },
        ]

        runTestCases({ requestBuilder, testCases })
      })
    })
  })
})
