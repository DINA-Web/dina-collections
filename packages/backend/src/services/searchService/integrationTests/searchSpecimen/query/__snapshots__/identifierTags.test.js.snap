// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchSpecimen - query - tags - identifier tag aggregations no filters returns all aggregated identifiers when no filters are present (dina) 1`] = `
Array [
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-500001",
      "tagType": "catalog-no",
      "tagValue": "500001",
      "type": "identifierTag",
    },
    "id": "catalog-no-500001",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-530183",
      "tagType": "catalog-no",
      "tagValue": "530183",
      "type": "identifierTag",
    },
    "id": "catalog-no-530183",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-534406",
      "tagType": "catalog-no",
      "tagValue": "534406",
      "type": "identifierTag",
    },
    "id": "catalog-no-534406",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-583124",
      "tagType": "catalog-no",
      "tagValue": "583124",
      "type": "identifierTag",
    },
    "id": "catalog-no-583124",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-585522",
      "tagType": "catalog-no",
      "tagValue": "585522",
      "type": "identifierTag",
    },
    "id": "catalog-no-585522",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-587520",
      "tagType": "catalog-no",
      "tagValue": "587520",
      "type": "identifierTag",
    },
    "id": "catalog-no-587520",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-590325",
      "tagType": "catalog-no",
      "tagValue": "590325",
      "type": "identifierTag",
    },
    "id": "catalog-no-590325",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-620285",
      "tagType": "catalog-no",
      "tagValue": "620285",
      "type": "identifierTag",
    },
    "id": "catalog-no-620285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-621445",
      "tagType": "catalog-no",
      "tagValue": "621445",
      "type": "identifierTag",
    },
    "id": "catalog-no-621445",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-628009",
      "tagType": "catalog-no",
      "tagValue": "628009",
      "type": "identifierTag",
    },
    "id": "catalog-no-628009",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-630096",
      "tagType": "catalog-no",
      "tagValue": "630096",
      "type": "identifierTag",
    },
    "id": "catalog-no-630096",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-805176",
      "tagType": "catalog-no",
      "tagValue": "805176",
      "type": "identifierTag",
    },
    "id": "catalog-no-805176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-825005",
      "tagType": "catalog-no",
      "tagValue": "825005",
      "type": "identifierTag",
    },
    "id": "catalog-no-825005",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-956051",
      "tagType": "catalog-no",
      "tagValue": "956051",
      "type": "identifierTag",
    },
    "id": "catalog-no-956051",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-985093",
      "tagType": "catalog-no",
      "tagValue": "985093",
      "type": "identifierTag",
    },
    "id": "catalog-no-985093",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-985729",
      "tagType": "catalog-no",
      "tagValue": "985729",
      "type": "identifierTag",
    },
    "id": "catalog-no-985729",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "loan-no-2012-21",
      "tagType": "loan-no",
      "tagValue": "2012-21",
      "type": "identifierTag",
    },
    "id": "loan-no-2012-21",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-1,285",
      "tagType": "old-skeleton-no",
      "tagValue": "1,285",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-1,285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-1; 4406",
      "tagType": "old-skeleton-no",
      "tagValue": "1; 4406",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-1; 4406",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-20",
      "tagType": "old-skeleton-no",
      "tagValue": "20",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-20",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-5176",
      "tagType": "old-skeleton-no",
      "tagValue": "5176",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-5176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-6",
      "tagType": "old-skeleton-no",
      "tagValue": "6",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-6",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-9",
      "tagType": "old-skeleton-no",
      "tagValue": "9",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-9",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-925",
      "tagType": "old-skeleton-no",
      "tagValue": "925",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-925",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-a825005",
      "tagType": "old-skeleton-no",
      "tagValue": "a825005",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-a825005",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-bly 45, 1008, 3124",
      "tagType": "old-skeleton-no",
      "tagValue": "bly 45, 1008, 3124",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-bly 45, 1008, 3124",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-1,285",
      "tagType": "old-skin-no",
      "tagValue": "1,285",
      "type": "identifierTag",
    },
    "id": "old-skin-no-1,285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-1; 4406; 52",
      "tagType": "old-skin-no",
      "tagValue": "1; 4406; 52",
      "type": "identifierTag",
    },
    "id": "old-skin-no-1; 4406; 52",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-20",
      "tagType": "old-skin-no",
      "tagValue": "20",
      "type": "identifierTag",
    },
    "id": "old-skin-no-20",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-321",
      "tagType": "old-skin-no",
      "tagValue": "321",
      "type": "identifierTag",
    },
    "id": "old-skin-no-321",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-5176",
      "tagType": "old-skin-no",
      "tagValue": "5176",
      "type": "identifierTag",
    },
    "id": "old-skin-no-5176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-9",
      "tagType": "old-skin-no",
      "tagValue": "9",
      "type": "identifierTag",
    },
    "id": "old-skin-no-9",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-925",
      "tagType": "old-skin-no",
      "tagValue": "925",
      "type": "identifierTag",
    },
    "id": "old-skin-no-925",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "sva-no-v0253/98",
      "tagType": "sva-no",
      "tagValue": "v0253/98",
      "type": "identifierTag",
    },
    "id": "sva-no-v0253/98",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "sva-no-v0615/98",
      "tagType": "sva-no",
      "tagValue": "v0615/98",
      "type": "identifierTag",
    },
    "id": "sva-no-v0615/98",
    "type": "customObject",
  },
]
`;

exports[`searchSpecimen - query - tags - identifier tag aggregations no filters returns all aggregated identifiers when no filters are present (raw) 1`] = `
Array [
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-500001",
      "tagType": "catalog-no",
      "tagValue": "500001",
      "type": "identifierTag",
    },
    "id": "catalog-no-500001",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-530183",
      "tagType": "catalog-no",
      "tagValue": "530183",
      "type": "identifierTag",
    },
    "id": "catalog-no-530183",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-534406",
      "tagType": "catalog-no",
      "tagValue": "534406",
      "type": "identifierTag",
    },
    "id": "catalog-no-534406",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-583124",
      "tagType": "catalog-no",
      "tagValue": "583124",
      "type": "identifierTag",
    },
    "id": "catalog-no-583124",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-585522",
      "tagType": "catalog-no",
      "tagValue": "585522",
      "type": "identifierTag",
    },
    "id": "catalog-no-585522",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-587520",
      "tagType": "catalog-no",
      "tagValue": "587520",
      "type": "identifierTag",
    },
    "id": "catalog-no-587520",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-590325",
      "tagType": "catalog-no",
      "tagValue": "590325",
      "type": "identifierTag",
    },
    "id": "catalog-no-590325",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-620285",
      "tagType": "catalog-no",
      "tagValue": "620285",
      "type": "identifierTag",
    },
    "id": "catalog-no-620285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-621445",
      "tagType": "catalog-no",
      "tagValue": "621445",
      "type": "identifierTag",
    },
    "id": "catalog-no-621445",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-628009",
      "tagType": "catalog-no",
      "tagValue": "628009",
      "type": "identifierTag",
    },
    "id": "catalog-no-628009",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-630096",
      "tagType": "catalog-no",
      "tagValue": "630096",
      "type": "identifierTag",
    },
    "id": "catalog-no-630096",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-805176",
      "tagType": "catalog-no",
      "tagValue": "805176",
      "type": "identifierTag",
    },
    "id": "catalog-no-805176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-825005",
      "tagType": "catalog-no",
      "tagValue": "825005",
      "type": "identifierTag",
    },
    "id": "catalog-no-825005",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-956051",
      "tagType": "catalog-no",
      "tagValue": "956051",
      "type": "identifierTag",
    },
    "id": "catalog-no-956051",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-985093",
      "tagType": "catalog-no",
      "tagValue": "985093",
      "type": "identifierTag",
    },
    "id": "catalog-no-985093",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "catalog-no-985729",
      "tagType": "catalog-no",
      "tagValue": "985729",
      "type": "identifierTag",
    },
    "id": "catalog-no-985729",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "loan-no-2012-21",
      "tagType": "loan-no",
      "tagValue": "2012-21",
      "type": "identifierTag",
    },
    "id": "loan-no-2012-21",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-1,285",
      "tagType": "old-skeleton-no",
      "tagValue": "1,285",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-1,285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-1; 4406",
      "tagType": "old-skeleton-no",
      "tagValue": "1; 4406",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-1; 4406",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-20",
      "tagType": "old-skeleton-no",
      "tagValue": "20",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-20",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-5176",
      "tagType": "old-skeleton-no",
      "tagValue": "5176",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-5176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-6",
      "tagType": "old-skeleton-no",
      "tagValue": "6",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-6",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-9",
      "tagType": "old-skeleton-no",
      "tagValue": "9",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-9",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-925",
      "tagType": "old-skeleton-no",
      "tagValue": "925",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-925",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-a825005",
      "tagType": "old-skeleton-no",
      "tagValue": "a825005",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-a825005",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skeleton-no-bly 45, 1008, 3124",
      "tagType": "old-skeleton-no",
      "tagValue": "bly 45, 1008, 3124",
      "type": "identifierTag",
    },
    "id": "old-skeleton-no-bly 45, 1008, 3124",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-1,285",
      "tagType": "old-skin-no",
      "tagValue": "1,285",
      "type": "identifierTag",
    },
    "id": "old-skin-no-1,285",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-1; 4406; 52",
      "tagType": "old-skin-no",
      "tagValue": "1; 4406; 52",
      "type": "identifierTag",
    },
    "id": "old-skin-no-1; 4406; 52",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-20",
      "tagType": "old-skin-no",
      "tagValue": "20",
      "type": "identifierTag",
    },
    "id": "old-skin-no-20",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-321",
      "tagType": "old-skin-no",
      "tagValue": "321",
      "type": "identifierTag",
    },
    "id": "old-skin-no-321",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-5176",
      "tagType": "old-skin-no",
      "tagValue": "5176",
      "type": "identifierTag",
    },
    "id": "old-skin-no-5176",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-9",
      "tagType": "old-skin-no",
      "tagValue": "9",
      "type": "identifierTag",
    },
    "id": "old-skin-no-9",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "old-skin-no-925",
      "tagType": "old-skin-no",
      "tagValue": "925",
      "type": "identifierTag",
    },
    "id": "old-skin-no-925",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "sva-no-v0253/98",
      "tagType": "sva-no",
      "tagValue": "v0253/98",
      "type": "identifierTag",
    },
    "id": "sva-no-v0253/98",
    "type": "customObject",
  },
  Object {
    "attributes": Object {
      "count": 1,
      "key": "sva-no-v0615/98",
      "tagType": "sva-no",
      "tagValue": "v0615/98",
      "type": "identifierTag",
    },
    "id": "sva-no-v0615/98",
    "type": "customObject",
  },
]
`;
